### DISCUSSION TITLE
Include Grouping for Activity and User in Timesheet Summary

### DISCUSSION THREAD

**Discussion by derdoc800** [kimai/kimai#4536] [Questions & Answers] [answered]
We have the requirement to show on the first Timesheet page the summary not only by customer and project but also by activity and by user. We tried a couple adjustment in the .pdf.twig file and were able to add the columns for activity and user but the reference object is summary and the output remains empty. 

![grafik](https://github.com/kimai/kimai/assets/155985205/d816fc1f-926e-4462-85fd-9d7e64f9308e)

Before we keep trying by trial and error does anybody have an idea if this is possible with the current underlying timesheet setup?

**Comment by kevinpapst** [comment, marked as answer]
This data is available: https://github.com/kimai/kimai/blob/main/src/Export/Base/RendererTrait.php

**Reply by derdoc800** [reply]
Thank you very much! Sample Syntax to get all users in the timesheet export is.

<table class="items">
    {% for id , summary in summaries %}
        <tr>
            <td>{{summary.customer}}</td>
            <td>{{summary.project}}</td>
        </tr>
        {% for id , user in summary.users %}
        <tr>
        <td>{{user.user}}</td>
        <td>{{user.duration|duration(decimal)}}</td>
        </tr>
        {% endfor %}
    {% endfor %}
 </table>

**Reply by kevinpapst** [reply]
Glad you could solve the rest by yourself and thanks for sharing the solution 👍
